### ISSUE TITLE
[ENH]: Set 'rotation' to clabel

### ISSUE THREAD

**Issue by auroralin0408** [matplotlib/matplotlib#28799]
### Problem

I am using `ax.clabel` but the result seemed not good, as the lebel perpendicular to the contour instead of aligning with it. 

here's my code and result.
`ax.clabel(cl,fontsize=10,inline=False,rightside_up=True,manual=((1.3e5,-150),(1.2e5,-130),(1.1e5,-130),(1.1e5,-90),(1.1e5,-50)))`

<img width="337" alt="image" src="https://github.com/user-attachments/assets/f79d13a4-032a-4c8f-8037-46132841ad29">

with `inline=True`:
<img width="339" alt="image" src="https://github.com/user-attachments/assets/cfe58a2f-c4b2-4aed-9245-a0ea437c1c8b">

### Proposed solution

would it be possible to set rotation for it?

**Comment by jklymak** [comment]
Please provide a minimal self contained example.  You have Alamo certainly called clabel too early and flipped an axes after the call

**Comment by auroralin0408** [comment]
You are right. I flipped X-axis after the call, and now I put the call at the end. Thanks for your timely reply!
